src/app/pages/Profile/Profile.tsx

```diff
@@ -0,0 +1,124 @@
+import axios from 'axios';
+import React, { useState } from 'react';
+import PasswordStrengthBar from 'react-password-strength-bar';
+import Button from '../../components/Button/Button';
+import Headline from '../../components/Headline/Headline';
+import Input from '../../components/Input/Input';
+import useCurrentUser from '../../hooks/useCurrentUser';
+import useExistsUser from '../../hooks/useExistsUser';
+import styles from './Profile.module.css';
+import cookies from 'js-cookie';
+import { useQueryClient } from 'react-query';
+
+export type ProfileProps = {
+  className?: string;
+};
+
+export default function Profile({ className = '' }: ProfileProps): JSX.Element {
+  const [username, setUsername] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const [usernameToCheck, setUsernameToCheck] = useState<string>('');
+  const [formError, setFormError] = useState<string | null>(null);
+
+  const currentUser = useCurrentUser();
+
+  const queryClient = useQueryClient();
+  const userExists = useExistsUser(usernameToCheck);
+  const loginMode = userExists === true;
+  const registerMode = userExists === false;
+
+  async function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+
+    const registerResult =
+      registerMode &&
+      (await axios.post('/api/user', {
+        user: {
+          username,
+          password,
+        },
+      }));
+
+    if (loginMode || (registerResult && registerResult.status === 201)) {
+      try {
+        await axios.post('/api/user/login', {
+          user: {
+            username,
+            password,
+          },
+        });
+        setUsername('');
+      } catch (error) {
+        setFormError('Wrong password');
+      }
+    }
+  }
+
+  function logout() {
+    cookies.remove('auth');
+
+    queryClient.setQueryData(['checkUser', currentUser], true);
```

Comment:
I don't understand why you only interact with `queryClient` on logout

Reply:
It deletes the cached value of the useExistsUser hook. It's for the scenario: I register a new user, login, and logout within one session. When I return to the register/login screen, I need the refetch.

Reply:
Replaced by `queryClient.invalidateQueries();`